docarray/index/backends/milvus.py

```diff
@@ -0,0 +1,328 @@
+
+TSchema = TypeVar('TSchema', bound=BaseDoc)
+
+
+class MilvusDocumentIndex(BaseDocIndex, Generic[TSchema]):
+    def __init__(self, db_config=None, index_name=None, **kwargs):
+        super().__init__(db_config=db_config, **kwargs)
+        self._db_config: MilvusDocumentIndex.DBConfig = cast(
+            MilvusDocumentIndex.DBConfig, self._db_config
+        )
+
+        self._client = connections.connect(
+            db_name="default",
+            host=self._db_config.host,
+            user=self._db_config.user,
+            password=self._db_config.password,
+            token=self._db_config.token,
+        )
+
+        self._db_config.index_name = index_name
+        self._validate_columns()
+        self._create_collection_name()
+        self._collection = self._init_index()
+        self._build_index()
+        self._logger.info(f"{self.__class__.__name__} has been initialized")
+
+    @dataclass
+    class DBConfig(BaseDocIndex.DBConfig):
+        collection_name: str = None
+        host: str = "localhost"
+        port: int = 19530
+        user: Optional[str] = ""
+        password: Optional[str] = ""
+        token: Optional[str] = ""
+        index_name: str = None
+        index_type: str = "IVF_FLAT"
+        index_metric: str = "L2"
+        index_params: Dict = field(default_factory=lambda: {"nlist": 1024})
+        default_column_config: Dict[Type, Dict[str, Any]] = field(
+            default_factory=lambda: defaultdict(dict)
+        )
+
+    def python_type_to_db_type(self, python_type: Type) -> Any:
+        type_map = {
+            int: DataType.INT64,
+            float: DataType.FLOAT,
+            str: DataType.VARCHAR,
+            bytes: DataType.VARCHAR,
+            np.ndarray: DataType.FLOAT_VECTOR,
+            list: DataType.FLOAT_VECTOR,
+            AnyTensor: DataType.FLOAT_VECTOR,
+            AbstractTensor: DataType.FLOAT_VECTOR,
+        }
+
+        if issubclass(python_type, ID):
+            return DataType.INT64  # Primary key must be int64
+
+        for py_type, db_type in type_map.items():
+            if safe_issubclass(python_type, py_type):
+                return db_type
+
+        raise ValueError("No corresponding milvus type")
+
+    def _init_index(self) -> Collection:
+        if not utility.has_collection(self._db_config.collection_name):
+            print(self._db_config.collection_name)
+            fields = [
+                FieldSchema(
+                    name="id",
+                    dtype=DataType.INT64,
+                    is_primary=True,
+                    auto_id=True,
+                )
+            ]  # Initiliaze the id
+            fields.extend(
+                [
+                    FieldSchema(
+                        name="doc_id" if column_name == "id" else column_name,
+                        dtype=DataType.VARCHAR if column_name == "id" else info.db_type,
+                        is_primary=False,
```

Comment:
shouldn't `id``be completely skipped here since it is alrady added to the schema above? Won't this add the id twice, once as `id` and once as `doc_id`?

Reply:
Thanks for the feedback! I'll only keep doc_id